src/main/java/org/broadinstitute/hellbender/transformers/BQSRReadTransformer.java

```diff
@@ -53,24 +55,47 @@ public BQSRReadTransformer(File bqsrRecalFile, int quantizationLevels, boolean d
     }
 
     /**
-     * Recalibrates the base qualities of a read
-     * <p>
-     * It updates the base qualities of the read with the new recalibrated qualities (for all event types)
-     * <p>
-     * Implements a serial recalibration of the reads using the combinational table.
-     * First, we perform a positional recalibration, and then a subsequent dinuc correction.
-     * <p>
-     * Given the full recalibration table, we perform the following preprocessing steps:
-     * <p>
-     * - calculate the global quality score shift across all data [DeltaQ]
-     * - calculate for each of cycle and dinuc the shift of the quality scores relative to the global shift
-     * -- i.e., DeltaQ(dinuc) = Sum(pos) Sum(Qual) Qempirical(pos, qual, dinuc) - Qreported(pos, qual, dinuc) / Npos * Nqual
-     * - The final shift equation is:
-     * <p>
-     * Qrecal = Qreported + DeltaQ + DeltaQ(pos) + DeltaQ(dinuc) + DeltaQ( ... any other covariate ... )
+     * Constructor using a GATK Report file
      *
-     * @param read the read to recalibrate
+     * @param recalInfo          the output of BaseRecalibration, containing the recalibration information
+     * @param quantizationLevels number of bins to quantize the quality scores
+     * @param disableIndelQuals  if true, do not emit base indel qualities
+     * @param preserveQLessThan  preserve quality scores less than this value
      */
+    public BQSRReadTransformer(final BaseRecalOutput recalInfo, int quantizationLevels, boolean disableIndelQuals, final int preserveQLessThan, final boolean emitOriginalQuals, final double globalQScorePrior) {
```

Comment:
quantization levels and disableIndelQuals should probably be final too for consistency